### PR TITLE
Issue:3897880:PDR exclusion list tests

### PR DESCRIPTION
**Pull request by vg12345**

### CODE REVIEW

File: plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+echo "Init PDR configuration file"
+CONFIG_FILE="/config/pdr_deterministic.conf"
+mkdir -p /config
+cp -f $PLUGIN_DIR/build/config/pdr_deterministic.conf "$CONFIG_FILE"
+sed -i -e 's/\nTEST_MODE=True\n//g' "$CONFIG_FILE"
+echo -e '\n[Common]\nTEST_MODE=True\n' >> "$CONFIG_FILE"
+sed -i -e 's/DRY_RUN=False/DRY_RUN=True/g' "$CONFIG_FILE"
+sed -i -e 's/INTERVAL=300/INTERVAL=10/g' "$CONFIG_FILE"
+sed -i -e 's/CONFIGURED_TEMP_CHECK=False/CONFIGURED_TEMP_CHECK=True/g' "$CONFIG_FILE"
```
Comment (drorlevy) [inline plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh:17]: also add the link down check to be true

File: plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+cp -r utils $PLUGIN_DIR/ufm_sim_web_service
+cp -r utils $PLUGIN_DIR/tests
+
+echo "Init PDR configuration file"
+CONFIG_FILE="/config/pdr_deterministic.conf"
+mkdir -p /config
+cp -f $PLUGIN_DIR/build/config/pdr_deterministic.conf "$CONFIG_FILE"
+sed -i -e 's/\nTEST_MODE=True\n//g' "$CONFIG_FILE"
+echo -e '\n[Common]\nTEST_MODE=True\n' >> "$CONFIG_FILE"
```
Comment (drorlevy) [inline plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh:14]: echoing without checking if it exists before, caused us trouble with running these tests multiple times previously.

File: plugins/pdr_deterministic_plugin/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+flask<=3.0.3
```
Comment (drorlevy) [inline plugins/pdr_deterministic_plugin/requirements.txt:1]: good that you added a requirements file, maybe also change it in the Dockerfile not to have it duplicated

File: plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py [hunk 4]
```diff
@@ -296,14 +324,17 @@ def validate_simulation_data():
     return True
 
 def check_logs(config):
+    """
+    Analize output log and create tests results
```
Comment (drorlevy) [inline plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py:328]: typo: analyze

File: plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh (hunk 2 above)
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh:14]: Line 
sed -i -e 's/\nTEST_MODE=\n//g' "$CONFIG_FILE"
will remove any existing TEST_MODE lines from the file

File: plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py (hunk 4 above)
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py]: Fixed

File: plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh (hunk 1 above)
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh:17]: Added

File: plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py [hunk 5]
```diff
@@ -245,23 +260,27 @@ def excluded_ports_simulation(endpoint):
...
+    Create an array of ports in size of ports_num
+    """
     ports_list = []
     ports_names = []
     for _ in range(ports_num):
         port_guid = f'0x{random.randrange(16**16):016x}'
         # holds the prefix of each simulated csv rows,
         # list of counters structures(will be filled further)
-        port_num = random.randint(1, 99)
+        port_num = random.randint(10, 99)
```
[the remark this answers is not in the document]
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py:273]: It does not matter. Just to make output look more pretty (aligned) in final report.

[the remark this answers is not in the document]
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py:273]: It does not matter. Just to make output look more pretty (aligned) in final report.

File: plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash -x
+
+PLUGIN_DIR="plugins/pdr_deterministic_plugin"
+pip install -r $PLUGIN_DIR/requirements.txt >/dev/null 2>&1
```
[the remark this answers is not in the document]
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh:4]: The packages installed first step are not available in the environment where 'run_pdr_standalone_pytest.sh' is executed with 'sudo'.

File: .github/workflows/pdr_plugin_ci_workflow.yml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        pkill -9 -f isolation_algo.py 2>/dev/null || true
+
+    - name: Run full simulation test
+      timeout-minutes: 10
+      run: |
+        sudo bash $PDRPATH/.pytest/run_pdr_standalone_pytest.sh
+        echo "Starting simulated test"
+        python $PDRPATH/tests/simulation_telemetry.py
+        echo "Terminating standalone PDR process"
+        pkill -9 -f isolation_algo.py 2>/dev/null || true
```
[the remark this answers is not in the document]
Comment (vg12345) [reply .github/workflows/pdr_plugin_ci_workflow.yml]: Fixed

File: plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+cp -f $PLUGIN_DIR/build/config/pdr_deterministic.conf "$CONFIG_FILE"
+sed -i -e 's/\nTEST_MODE=True\n//g' "$CONFIG_FILE"
+echo -e '\n[Common]\nTEST_MODE=True\n' >> "$CONFIG_FILE"
+sed -i -e 's/DRY_RUN=False/DRY_RUN=True/g' "$CONFIG_FILE"
+sed -i -e 's/INTERVAL=300/INTERVAL=10/g' "$CONFIG_FILE"
+sed -i -e 's/CONFIGURED_TEMP_CHECK=False/CONFIGURED_TEMP_CHECK=True/g' "$CONFIG_FILE"
+sed -i -e 's/DEISOLATE_CONSIDER_TIME=5/DEISOLATE_CONSIDER_TIME=1/g' "$CONFIG_FILE"
+
+"Terminating standalone PDR process"
+pkill -9 -f isolation_algo.py 2>/dev/null || true
```
[the remark this answers is not in the document]
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/.pytest/run_pdr_standalone_pytest.sh]: Fixed

**Review by egershonNvidia — APPROVED** [review]

**Review by drorlevy — APPROVED** [review]

**Pull request merged**